client/src/visualiser-src/graph-visualiser/util/util.ts

```diff
@@ -0,0 +1,321 @@
+/* eslint-disable */
+// TODO: remove the eslint disable
+
+import * as d3 from 'd3';
+import {
+  CONTAINER_DEFAULT_HEIGHT,
+  CONTAINER_DEFAULT_WIDTH,
+  NODE_DIAMETER,
+  STROKE_WIDTH,
+  VISUALISER_CANVAS,
+} from 'visualiser-src/common/constants';
+import {
+  ARROWHEAD_SIZE_FACTOR,
+  EDGE_ATTRACTIVE_FORCE_MULTIPLIER,
+  EDGE_WIDTH,
+  INTER_VERTEX_FORCE,
+  VERTEX_FONT_SIZE,
+  WEIGHT_LABEL_SIZE,
+} from 'visualiser-src/common/constants';
+
+function getPrimitiveVal(value) {
+  return value !== null && typeof value === 'object' ? value.valueOf() : value;
+}
+
+// Defining the arrowhead for directed edges.
+// Sourced the attributes from here: http://bl.ocks.org/fancellu/2c782394602a93921faff74e594d1bb1.
+// TODO: these could be defined declaratively elsewhere
+function defineArrowheads() {
+  const graph = d3.select(VISUALISER_CANVAS);
+
+  graph
+    .append('defs')
+    .append('marker')
+    .attr('id', 'end-arrowhead')
+    .attr('viewBox', '-0 -10 20 20')
+    .attr('refX', NODE_DIAMETER / 2 - parseInt(EDGE_WIDTH) * 0.5) // The offset of the arrowhead.
+    .attr('refY', 0)
+    .attr('orient', 'auto')
+    .attr('markerWidth', (ARROWHEAD_SIZE_FACTOR * 1.6) / (parseInt(EDGE_WIDTH) / 5))
```

Comment:
Use markerUnits attribute to get rid of random scaling factor.

Reply:
Done.